discord/ext/commands/converter.py

```diff
@@ -341,7 +341,11 @@ def _get_id_matches(ctx, argument):
         if not match:
             raise MessageNotFound(argument)
         data = match.groupdict()
-        channel_id = discord.utils._get_as_snowflake(data, 'channel_id')
+        channel_id = data.get('channel_id')
+        if channel_id is None:
+            channel_id = ctx.channel and ctx.channel.id
```

Comment:
Are you sure that this is correct

Reply:
I've tested it with all the different types of message IDS (URL, channelid-messageid, messageid), as well as with cached and non-cached messages, as well as some thread messages.